src/components/Labels.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useState,
+  useEffect,
+} from "react";
+import { Theme } from "@material-ui/core/styles";
+
+import {
+  Collapse,
+  IconButton,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemSecondaryAction,
+  ListItemText,
+} from "@material-ui/core";
+import AddIcon from "@material-ui/icons/Add";
+import BackspaceIcon from "@material-ui/icons/Backspace";
+import ExpandLess from "@material-ui/icons/ExpandLess";
+import ExpandMore from "@material-ui/icons/ExpandMore";
+import LibraryAddIcon from "@material-ui/icons/LibraryAdd";
+
+import { Annotations } from "../annotation";
+
+interface Props {
+  annotationObject: Annotations;
+  presetLabels: string[];
+  theme: Theme;
+}
+
+export const Labels: FunctionComponent<Props> = ({
+  annotationObject,
+  presetLabels,
+  theme,
+}): ReactElement => {
+  const getMenuLabels = (): string[] => {
+    return presetLabels.filter((label) => !assignedLabels.includes(label));
+  };
+
+  const handleAddLabel = (label: string) => (): void => {
+    // Add a label to active annotation object and update some states.
+    annotationObject.addLabel(label);
+    setAssignedLabels(annotationObject.getLabels());
+  };
+
+  const handleRemoveLabel = (label: string) => (): void => {
+    // Remove a label from active annotation object and update some states.
+    annotationObject.removeLabel(label);
+    setAssignedLabels(annotationObject.getLabels());
+  };
+
+  useEffect(() => {
+    // Update menuLabels after assignedLabels has been reset.
+    setMenuLabels(getMenuLabels());
+  });
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const [assignedLabels, setAssignedLabels] = useState(
+    annotationObject.getLabels()
+  );
+  const [menuLabels, setMenuLabels] = useState(getMenuLabels());
```

Comment:
Am I right in thinking you're using `assignedLabels` and `menuLabels` as state in this component purely to make sure that it re-renders when labels are added or removed? Because `annotationsObject` is not a react component and so doesn't trigger a re-render when you call `annotationsObject.addLabel()`? If so, I wonder if we could improve this by making the annotationsObject part of `ui` state, then when you call `addLabel` it'd `ui` to re-render, passing the down the new annotationsObject to the canvases and to this component, thus triggering a re-render here.

Reply:
(not gonna block the merge for this of course, it should be a separate PR if we do it)

Reply:
Yes, you are right, these variables are there so that it the lists of variables re-renders properly. The reason why I did not define these as state in ui is to have a modular Labels.tsx. Also, these variables are not needed at the level of ui, i think?

Reply:
annotationsObject isn't _needed_ by `ui` per se, but it does already have it as a private property, just not as react state. There are lots of state elements that `ui` has but doesn't use itself, just passes them down to components as props whenever they change. I'm saying if we made annotationsObject part of `ui`'s react state rather than just private state, it might trigger re-renders automatically whenever we call `addLabel` or something on it from another component. I don't know if this would work though, normally you modify state with `setState`, not sure what happens if some state is an object like `annotationsObject` and you call a method on it directly.